lambdas/build-cri-oauth-request/src/main/java/uk/gov/di/ipv/core/buildcrioauthrequest/BuildCriOauthRequestHandler.java

```diff
@@ -397,70 +393,63 @@ private EvidenceRequest getEvidenceRequestForKbvCri(Vot targetVot) {
-                    JsonNode credentialSubject =
-                            OBJECT_MAPPER
-                                    .readTree(
-                                            SignedJWT.parse(vc.getVcString())
-                                                    .getPayload()
-                                                    .toString())
-                                    .path(VC_CLAIM)
-                                    .path(VC_CREDENTIAL_SUBJECT);
-                    if (credentialSubject.isMissingNode()) {
-                        LOGGER.error(
-                                LogHelper.buildLogMessage(
-                                        ErrorResponse.CREDENTIAL_SUBJECT_MISSING.getMessage()));
-                        throw new HttpResponseExceptionWithErrorBody(
-                                500, ErrorResponse.CREDENTIAL_SUBJECT_MISSING);
-                    }
-
-                    SharedClaims credentialsSharedClaims =
-                            OBJECT_MAPPER.readValue(
-                                    credentialSubject.toString(), SharedClaims.class);
-                    if (ADDRESS.equals(vc.getCri())) {
-                        hasAddressVc = true;
-                        sharedClaimsSet.forEach(sharedClaims -> sharedClaims.setAddress(null));
-                    } else if (hasAddressVc) {
-                        credentialsSharedClaims.setAddress(null);
-                    }
-                    verifyForAllowedSharedClaimAttrs(
-                            credentialsSharedClaims, criAllowedSharedClaimAttrs);
-                    sharedClaimsSet.add(credentialsSharedClaims);
+            if (VcHelper.isSuccessfulVc(vc)) {
+                var credentialSubject = vc.getCredential().getCredentialSubject();
+
+                if (credentialSubject instanceof PersonWithIdentity personWithIdentity
+                        && personWithIdentity.getName() != null) {
+                    sharedClaims.getName().addAll(personWithIdentity.getName());
+                }
+                if (credentialSubject instanceof PersonWithIdentity personWithIdentity
+                        && personWithIdentity.getBirthDate() != null) {
+                    sharedClaims.getBirthDate().addAll(personWithIdentity.getBirthDate());
+                }
+                if (ADDRESS.equals(vc.getCri())
+                        && credentialSubject instanceof AddressAssertion addressAssertion
+                        && addressAssertion.getAddress() != null) {
+                    sharedClaims.getAddress().addAll(addressAssertion.getAddress());
+                }
+                // TODO: should update tests to avoid needing this
+                if (ADDRESS.equals(vc.getCri())
+                        && credentialSubject instanceof IdentityCheckSubject addressAssertion) {
+                    sharedClaims.setAddress(
+                            addressAssertion.getAddress().stream().collect(Collectors.toSet()));
```

Comment:
Intellij is suggesting this can be simlified with:

`new HashSet<>(addressAssertion.getAddress())`